var/spack/repos/builtin/packages/py-spython/package.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class PySpython(PythonPackage):
+    """The Python API for working with Singularity containers."""
+
+    homepage = "https://github.com/singularityhub/singularity-cli"
+    pypi = "spython/spython-0.2.14.tar.gz"
+
+    version("0.2.14", sha256="49e22fbbdebe456b27ca17d30061489db8e0f95e62be3623267a23b85e3ce0f0")
+
+    variant(
+        "runtime",
+        default="none",
+        description="Container runtime installed by Spack for this package",
+        values=("none", "singularityce", "singularity"),
+        multi=False,
+    )
+
+    depends_on("singularityce@3.5.2:", when="runtime=singularityce", type="run")
+    depends_on("singularity@3.5.2:", when="runtime=singularity", type="run")
```

Comment:
Should we make a virtual provider for singularity so we don't need this complexity?

Reply:
Good point. Seems out of scope for this PR though; I would suggest leaving it to a separate one. 

Opening a github issue about it: #34386

Reply:
Not necessarily out of scope, could even be fewer lines of code than all of the multi-value variant stuff. Just a suggestion, not a requirement.

Reply:
The reason why I prefer to keep this aspect distinct is that I don't have the required knowledge of these two projects, to be able to guarantee that their APIs/user interfaces are, and crucially will be in the future, compatible with each other.  
Singularity is stale, as it has forked into SingularityCE and Apptainer, and I am concerned that future developments of SingularityCE may make it not compatible with the original Singularity.